clo-workflows/Scripts/src/service/dataService/SpDataService.ts

```diff
@@ -123,29 +124,55 @@ export class SpDataService implements IDataService {
             .get(this.cloRequestElementParser)
     }
 
-    fetchClientCompletedProjects(): Promise<Array<CloRequestElement>> {
+    async fetchClientCompletedProjects(): Promise<Array<CloRequestElement>> {
         return Promise.resolve(null)
     }
 
-    
-    fetchClientProjects(): Promise<Array<CloRequestElement>> {
-        return Promise.resolve(null)
+    async fetchClientProjects(): Promise<Array<CloRequestElement>> {
+        const clientProjects: Array<CloRequestElement> = await this.getHostWeb()
+            .lists.getByTitle(ListName.PROJECTS)
+            .items.get(this.cloRequestElementParser)
+        return clientProjects
+    }
+    async fetchClientProcesses(): Promise<Array<CloRequestElement>> {
+        const clientProcesses: Array<CloRequestElement> = await this.getHostWeb()
+            .lists.getByTitle(ListName.PROCESSES)
+            .items.get(this.cloRequestElementParser)
+        return clientProcesses
+    }
+    async fetchWorks(): Promise<Array<CloRequestElement>> {
+        const works: Array<CloRequestElement> = await this.getHostWeb()
+            .lists.getByTitle(ListName.WORKS)
+            .items.get(this.cloRequestElementParser)
+        return works
     }
-
     async createNote(note: INote, listName: ListName): Promise<void> {
         await this.getHostWeb()
             .lists.getByTitle(listName)
-            .items.add(note)        
+            .items.add(note)
+    }
+    async createProject(projectData: {}): Promise<ItemAddResult> {
```

Comment:
for createProject, createProcess, and createWork, having the parameter be CloRequestElement is more typesafe than just plain {} because it ensures that we only send stuff to the server that the server is expecting.